abd-clam/src/core/graph/_graph.rs

```diff
@@ -218,200 +246,284 @@ impl<'a, T: Number, U: Number, D: Dataset<T, U>> Graph<'a, T, U, D> {
+    ///
+    /// This method returns a reference to the distance matrix of the graph. The distance matrix
+    /// should be computed by calling the `with_distance_matrix` method before using this function.
+    #[allow(unused_mut, unused_variables, dead_code)]
+    pub const fn distance_matrix(&self) -> &[Vec<U>] {
+        todo!()
 
-    #[allow(dead_code)]
-    pub fn distance_matrix(&self) -> &[Vec<U>] {
-        self.distance_matrix
-            .as_ref()
-            .expect("Please call `with_distance_matrix` on the Graph before using `distance_matrix`.")
+        // self.distance_matrix
+        //     .as_ref()
+        //     .expect("Please call `with_distance_matrix` on the Graph before using `distance_matrix`.")
     }
 
-    #[allow(dead_code)]
-    pub fn adjacency_matrix(&self) -> &[Vec<bool>] {
-        self.adjacency_matrix
-            .as_ref()
-            .expect("Please call `with_adjacency_matrix` on the Graph before using `adjacency_matrix`.")
-    }
+    /// Returns a reference to the adjacency matrix of the graph.
+    ///
+    /// This method returns a reference to the adjacency matrix of the graph. The adjacency matrix
+    /// should be computed by calling the `with_adjacency_matrix` method before using this function.
+    #[allow(unused_mut, unused_variables, dead_code)]
+    pub const fn adjacency_matrix(&self) -> &[Vec<bool>] {
+        todo!()
 
-    #[allow(dead_code)]
-    pub fn diameter(&'a self) -> usize {
-        self.clusters
-            .iter()
-            .map(|&c| self.unchecked_eccentricity(c))
-            .max()
-            .unwrap()
+        // self.adjacency_matrix
+        //     .as_ref()
+        //     .expect("Please call `with_adjacency_matrix` on the Graph before using `adjacency_matrix`.")
     }
 
-    fn assert_contains(&self, c: &Cluster<T, U, D>) -> Result<(), String> {
+    /// Calculates and returns the diameter of the graph.
+    ///
+    /// The diameter is the maximum eccentricity value among all clusters in the graph.
+    /// It represents the longest shortest path between any two clusters in the graph.
+    #[allow(unused_mut, unused_variables, dead_code)]
+    pub const fn diameter(&'a self) -> usize {
+        todo!()
+        // self.clusters
+        //     .iter()
+        //     .map(|&c| self.unchecked_eccentricity(c))
+        //     .max()
+        //     .unwrap()
+    }
+
+    /// Asserts whether a given cluster is contained within the graph.
+    #[allow(unused_mut, unused_variables, dead_code)]
+    fn assert_contains(&self, c: &Cluster<U>) -> Result<(), String> {
         if self.clusters.contains(&c) {
             Ok(())
         } else {
             Err(format!("Cluster {c} is not in this graph."))
         }
     }
 
-    #[allow(dead_code)]
-    pub fn unchecked_vertex_degree(&'a self, c: &Cluster<T, U, D>) -> usize {
+    /// Returns the vertex degree (number of neighbors) of a given cluster.
+    /// Panics:
```

Comment:
There should be an empty line before the `Panics:` docs start. This should look like:

```rust
/// A one-line description of the function.
///
/// A longer description
/// of the function.
///
/// # Arguments
///
/// * `arg_1`: The first argument.
/// * `arg_2`: The second argument.
///
/// # Returns:
///
/// If there is any return value.
///
/// # Errors
///
/// If the return is a `Result` type
///
/// # Panics
///
/// * First panic condition.
/// * Second panic condition.
fn fancty_function(...) -> ... { ... }
```

Reply:
I refactored the documentation in graph.rs to match that template. Still need to work on the other files.